libs/core/include/Event.hpp

```diff
@@ -41,6 +41,25 @@ class Event {
     exit(1);
   }
 
+  inline int GetNObjectsInCollection(std::string name) {
```

Comment:
I tried it first but it did not work, it returned much larger values (like 9999) so I did this for now

Reply:
Ok, so this 9999 comes from a variable called `maxCollectionElements` defined in Helpers.hpp. I didn't figure out a way to have dynamic collections, so they are actually always of this fixed length. 

However, the vector which is stored in the event (e.g. muons, or electrons) is not std::vector, but a wrapper on top of it called "Collection", which allows to modify what `end()` function returns. This allows it to keep the fixed size in memory, but when you iterate over it, it will stop after reaching the last actual element.

In various places I was using GetCollectionSize and it seemed to work fine, but if you find a case where it doesn't work correctly and returns 9999, we should have a look and fix it.